rs/ethereum/cketh/docs/ckerc20.adoc

```diff
@@ -113,35 +113,35 @@ The ckERC20 implementation consists of a suite of canisters:
-   │                                 │                                     │──────────────────────────────────>│
-   │                                 │  mint(token_id, amount, principal)  │                                   │
-   │<──────────────────────────────────────────────────────────────────────────────────────────────────────────│
- ┌─┴──┐                      ┌───────┴───────┐                     ┌───────┴───────┐                        ┌──┴───┐
- │User│                      │ERC-20 Contract│                     │Helper Contract│                        │Minter│
- └────┘                      └───────────────┘                     └───────────────┘                        └──────┘
+ ┌────┐                      ┌───────────────┐                                      ┌───────────────┐                                                      ┌──────┐
+ │User│                      │ERC-20 Contract│                                      │Helper Contract│                                                      │Minter│
+ └─┬──┘                      └───────┬───────┘                                      └───────┬───────┘                                                      └──┬───┘
+   │                                 │                                                      │                                                                 │
+   │approve(helper_contract, amount) │                                                      │                                                                 │
+   │────────────────────────────────>│                                                      │                                                                 │
+   │                                 │                                                      │                                                                 │
+   │                depositErc20(token_id, amount, principal, subaccount)                   │                                                                 │
+   │───────────────────────────────────────────────────────────────────────────────────────>│                                                                 │
+   │                                 │           transferFrom(user, minter, amount)         │                                                                 │
+   │                                 │<─────────────────────────────────────────────────────│                                                                 │
+   │                                 │                                                      │                                                                 │
+   │                                 │                                                      │       get_events                                                │
+   │                                 │                                                      │<────────────────────────────────────────────────────────────────│
```

Comment:
in the code this event is emitted
```
 emit ReceivedEthOrErc20(
            erc20Address,
            msg.sender,
            amount,
            principal,
            subaccount
        );
```

but the docstring mentions 
```
   /**
     * @dev Emits the `ReceivedErc20` event if the transfer succeeds.
     */
```
this is also shown on etherscan in the "write contract" section

<img width="1342" alt="Screenshot 2024-12-06 at 16 32 15" src="https://github.com/user-attachments/assets/2485c673-77d2-4928-b5b1-0afc06336a37">